README.rst

```diff
@@ -50,6 +50,72 @@ If you wish to install ``modelmeta`` manually, follow the steps below.
     $ cd modelmeta
     $ pipenv install # --dev for development packages
 
+
+Adding new datafiles to a PCIC modelmeta database
+=================================================
+
+This repository contains two convenient scripts that add data files to an existing modelmeta database so that our websites can access data from them. They are installed when the package is installed.
+
+``index_netcdf`` adds one or more netCDF climate data files to a PCIC modelmeta-format database::
+
+  index_netcdf -d postgresql://username:password@monsoon.pcic.uvic.ca/database /path/to/files/*.nc
+
+Usernames and passwords can be found in Team Password Manager. To add files to the data portal, use database ``pcic_meta``; to add files to climate explore or plan2adapt, use database ``ce_meta_12f290b63791``.
+
+In order to determine the metadata of the file, the ``index_metadata`` script scans its netCDF attributes. If the file does not have all the `required attributes <https://pcic.uvic.ca/confluence/display/CSG/PCIC+metadata+standard+for+downscaled+data+and+hydrology+modelling+data>`_ specified, the ``index_metadata`` script will be unable to proceed. You can look at a file's attributes with the command::
+
+  ncdump -h file.nc
+
+
+and update attributes using the ``update_metadata`` script in the `climate-explorer-data-prep <https://github.com/pacificclimate/climate-explorer-data-prep>`_ respository. If you update file attributes, log your update YAML and a list of which files you used with in in the `data-prep-actions <https://github.com/pacificclimate/data-prep-actions>`_ repository, in case you need to reprocess or check the files later.
+
+Once files have been indexed into the database, they need to be associated to individual ensembles; each ensemble is associated with a particular project or website and contains all data files needed to support the functioning of that component. In most cases, a file will be added to more than one ensemble::
+
+  associate_ensemble -n ensemble_name -v 1 -d postgresql://username:password@monsoon.pcic.uvic.ca/database /path/to/files/*.nc
+
+An incomplete list of ensembles you might want to add a file to:
```

Comment:
Maybe instead of enumerating many of the possible ensembles, it would be better to give the reader the tools to find all of the possible ensembles? E.g.:

```sql
SELECT ensemble_name, ensemble_description FROM ensembles;
```

Reply:
My goal was to indicate which ensembles were in active use, and pitfalls about using them. The list is probably going to be out of date pretty much immediately, though, so maybe just including the SQL is better.

Reply:
Plus:

```bash
$ grep "ensemble_name ="  ~/code/git/pdp/pdp/portals/*.py
/pdp_code/pdp/portals/bccaq2_downscale.py:ensemble_name = 'bccaq_version_2'
/pdp_code/pdp/portals/bccaq_extremes.py:ensemble_name = 'bccaq_extremes'
/pdp_code/pdp/portals/bc_prism.py:ensemble_name = 'bc_prism_with_monthlies'
/pdp_code/pdp/portals/downscale_archive.py:ensemble_name = 'downscaled_gcms_archive'
/pdp_code/pdp/portals/gridded_observations.py:ensemble_name = 'gridded-obs-met-data'
/pdp_code/pdp/portals/vic_gen1.py:ensemble_name = 'vic_gen1'
```

Reply:
Hm, that makes sense. I'll provide an explicit list of the three "special" ensembles - `downscaled_canada`, `all_files`, and `p2a_rules`, and other than that, instructions on how to tell which ones the portals are currently using.